### ISSUE TITLE
--max-chat-history-tokens not respected

### ISSUE THREAD

**Issue by distributev** [Aider-AI/aider#1963]
### Issue

I started 'aider' with 

aider --max-chat-history-tokens **_8192_**

and 

/tokens 

shows

$ 0.0323   **_10,783_** chat history                                                              use /clear to clear

P.S - Claude Sonnet is the best model for me but is expensive. I tracked the ever growing  chat history  as one of the main culprits of the high costs with sonnet so I added  --max-chat-history-tokens **_8192_** to limit the history.
However I find the chat history more than the limit I gave it.

### Version and model info

Latest version of aider with Claude sonnet

**Comment by paul-gauthier** [comment]
Thanks for trying aider and filing this issue.

`--max-chat-history-tokens` controls when aider will try and summarize the chat history. It is not a hard limit.

**Comment by distributev** [comment]
Hi Paul, 

aider is the only AI code agent I find myself using it again and again. Others I install, try a bit but none sticks for longer time usage.

Coming back to max-chat-history-tokens

I find the history 4x times bigger than the configuration I gave. Is this expected? I understand it could be 10%, 20% or even 50% bigger than the configuration but 4x times? The reason I use that argument is to save on Claude costs

![image](https://github.com/user-attachments/assets/6e059da7-a0ed-4d41-93e2-fc3ee6f8ff26)

Another tangential related thing

For me caching seems to increase costs much more because I see lots 
of "write cache" for thousands of tokens each but almost no "cache hits" 
(or very few)

so I commented out # aider --cache-prompts --no-stream 

and tried with 

--max-chat-history-tokens
